### PR TITLE
[baresip-libre] Add new port

### PR DESCRIPTION
**Pull request by alfredh**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html)
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@alfredh, Thanks for your PR, I tested the usage locally failed by `libre:x64-windows` but get this error.
Please take a look:
````
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\v-zhli17\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_C_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe" -DCMAKE_CXX_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/libre/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "C:\Users\v-zhli17\source\repos\usageTest" 2>&1"
1> Working directory: C:\Users\v-zhli17\source\repos\usageTest\out\build\x64-Debug
1> [CMake] -- The C compiler identification is MSVC 19.36.32532.0
1> [CMake] -- The CXX compiler identification is MSVC 19.36.32532.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] CMake Error at F:/Feature-test/libre/vcpkg/scripts/buildsystems/vcpkg.cmake:853 (_find_package):
1> [CMake]   Could not find a package configuration file provided by "libre" with any of
1> [CMake]   the following names:
1> [CMake] 
1> [CMake]     libreConfig.cmake
1> [CMake]     libre-config.cmake
1> [CMake] 
1> [CMake]   Add the installation prefix of "libre" to CMAKE_PREFIX_PATH or set
1> [CMake]   "libre_DIR" to a directory containing one of the above files.  If "libre"
1> [CMake]   provides a separate development package or SDK, be sure it has been
1> [CMake]   installed.
1> [CMake] Call Stack (most recent call first):
1> [CMake]   usageTest/CMakeLists.txt:6 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
1> 'C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\v-zhli17\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_C_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe" -DCMAKE_CXX_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/libre/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "C:\Users\v-zhli17\source\repos\usageTest" 2>&1"' execution failed with error: ''C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Ninja"  -DCMAKE_BUILD_TYPE:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\v-zhli17\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_C_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe" -DCMAKE_CXX_COMPILER:FILEPATH="C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/libre/vcpkg/scripts/buildsystems/vcpkg.cmake -DCMAKE_MAKE_PROGRAM="C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\Ninja\ninja.exe" "C:\Users\v-zhli17\source\repos\usageTest" 2>&1"' returned with exit code: 1'.

````

CMakeLists.txt:
````
	 cmake_minimum_required(VERSION 3.18)
	
	 project(versionstest2)
	
	 find_package(libre CONFIG REQUIRED)

	 add_executable(main usageTest.cpp)

	 target_link_libraries(main PRIVATE libre::re)
````

**Comment by JonLiu1993** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

**Comment by alfredh** [comment]
here are my files after "vcpkg install libre":


```
alfredh@debian:~/tmp/vcpkg$ tree installed/x64-linux/share/libre/
installed/x64-linux/share/libre/
├── copyright
├── re-config.cmake
├── usage
├── vcpkg_abi_info.txt
└── vcpkg.spdx.json

0 directories, 5 files




alfredh@debian:~/tmp/vcpkg$ ./vcpkg owns libre
libre:x64-linux: x64-linux/debug/lib/libre.a
libre:x64-linux: x64-linux/debug/lib/pkgconfig/libre.pc
libre:x64-linux: x64-linux/lib/libre.a
libre:x64-linux: x64-linux/lib/pkgconfig/libre.pc
libre:x64-linux: x64-linux/share/libre/copyright
libre:x64-linux: x64-linux/share/libre/re-config.cmake
libre:x64-linux: x64-linux/share/libre/usage
libre:x64-linux: x64-linux/share/libre/vcpkg.spdx.json
libre:x64-linux: x64-linux/share/libre/vcpkg_abi_info.txt
```

**Comment by dg0yt** [comment]
> x64-linux/share/libre/re-config.cmake

This is not going to work.
- For package name `re`: It won't be found in dir `libre`.
- For package name `libre`: It won't be found with the `re-config.cmake` name.

> ~~~cmake
> vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/re)
> ~~~

So the config comes from dir `re`. It needs to go dir `re`:

~~~cmake
vcpkg_cmake_config_fixup(CONFIG_PATH lib/cmake/re PACKAGE_NAME re)
~~~

Yes, the parameter is `PACKAGE_NAME`, but in fact it is the output directory name. The default value is the port name. `vcpkg_cmake_config_fixup` is a little bit stupid, at least until #31053.

**Comment by alfredh** [comment]
hi, thanks for your comment. I updated the code as suggested.

**Comment by dg0yt** [comment]
Don't forget to mark the PR as ready for review when you are done.

**Comment by JonLiu1993** [comment]
@alfredh, I tested usage again but also failed:

CMakeLists:
```
cmake_minimum_required(VERSION 3.16)
project(min-vcpkg-tiff-test)


find_package(libre CONFIG REQUIRED)

add_executable(main usageTest.cpp)

target_link_libraries(main PRIVATE libre::re)

```
````
1> CMake generation started for configuration: 'x64-Debug'.
1> Command line: "C:\WINDOWS\system32\cmd.exe" /c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/test/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\usageTest" 2>&1"
1> Working directory: C:\Users\test\source\repos\usageTest\out\build\x64-Debug
1> [CMake] -- Selecting Windows SDK version 10.0.22621.0 to target Windows 10.0.19045.
1> [CMake] -- The C compiler identification is MSVC 19.36.32532.0
1> [CMake] -- The CXX compiler identification is MSVC 19.36.32532.0
1> [CMake] -- Detecting C compiler ABI info
1> [CMake] -- Detecting C compiler ABI info - done
1> [CMake] -- Check for working C compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting C compile features
1> [CMake] -- Detecting C compile features - done
1> [CMake] -- Detecting CXX compiler ABI info
1> [CMake] -- Detecting CXX compiler ABI info - done
1> [CMake] -- Check for working CXX compiler: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.36.32532/bin/Hostx64/x64/cl.exe - skipped
1> [CMake] -- Detecting CXX compile features
1> [CMake] -- Detecting CXX compile features - done
1> [CMake] CMake Error at F:/test/vcpkg/scripts/buildsystems/vcpkg.cmake:853 (_find_package):
1> [CMake]   Could not find a package configuration file provided by "libre" with any of
1> [CMake]   the following names:
1> [CMake] 
1> [CMake]     libreConfig.cmake
1> [CMake]     libre-config.cmake
1> [CMake] 
1> [CMake]   Add the installation prefix of "libre" to CMAKE_PREFIX_PATH or set
1> [CMake]   "libre_DIR" to a directory containing one of the above files.  If "libre"
1> [CMake]   provides a separate development package or SDK, be sure it has been
1> [CMake]   installed.
1> [CMake] Call Stack (most recent call first):
1> [CMake]   usageTest/CMakeLists.txt:5 (find_package)
1> [CMake] -- Configuring incomplete, errors occurred!
1> 'C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/test/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\usageTest" 2>&1"' execution failed with error: ''C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/test/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\usageTest" 2>&1"' returned with exit code: 1'.

````
image

image

File: ports/libre/usage [hunk 1]
```diff
@@ -0,0 +1,4 @@
+The package libre is compatible with built-in CMake targets:
+
+    find_package(libre CONFIG REQUIRED)
```
Comment (dg0yt) [inline ports/libre/usage:3]: It needs to be fixed here.

File: ports/libre/usage [hunk 2]
```diff
@@ -0,0 +1,4 @@
...
+    target_link_libraries(main PRIVATE libre::re)
```
Comment (dg0yt) [inline ports/libre/usage:4]: Which namespace does it use?

File: ports/libre/usage (hunk 1 above)
Comment (JonLiu1993) [reply ports/libre/usage:3]: Yes, when I try to use `find_package(re CONFIG REQUIRED)` it solves the find --package problem, but it suffers from the namespace problem you noticed next:
````

1> [CMake] -- Configuring done (25.2s)
1> [CMake] CMake Error at usageTest/CMakeLists.txt:9 (target_link_libraries):
1> [CMake]   Target "main" links to:
1> [CMake] 
1> [CMake]     libre::re
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
1> [CMake] 
1> [CMake] 
1> [CMake] 
1> [CMake] -- Generating done (0.1s)
1> [CMake] CMake Generate step failed.  Build files cannot be regenerated correctly.
1> 'C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/libre/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\usageTest" 2>&1"' execution failed with error: ''C:\WINDOWS\system32\cmd.exe' '/c "%SYSTEMROOT%\System32\chcp.com 65001 >NUL && "C:\PROGRAM FILES\MICROSOFT VISUAL STUDIO\2022\COMMUNITY\COMMON7\IDE\COMMONEXTENSIONS\MICROSOFT\CMAKE\CMake\bin\cmake.exe"  -G "Visual Studio 17 2022" -A x64  -DCMAKE_CONFIGURATION_TYPES:STRING="Debug" -DCMAKE_INSTALL_PREFIX:PATH="C:\Users\test\source\repos\usageTest\out\install\x64-Debug" -DCMAKE_TOOLCHAIN_FILE=F:/Feature-test/libre/vcpkg/scripts/buildsystems/vcpkg.cmake "C:\Users\test\source\repos\usageTest" 2>&1"' returned with exit code: 1'.

````

**Comment by dg0yt** [comment]
Oh, re-config.cmake isn't exported CMake config. You need help with CMake ...

**Comment by dg0yt** [comment]
... instead of review comments:
https://github.com/alfredh/vcpkg/pull/1

**Comment by alfredh** [comment]
Many thanks for your patch, I have applied it to this PR.

There are some checks failing, currently around 5 failing checks, including
arm64_android, x64_windows etc...

**Review by dg0yt — CHANGES_REQUESTED** [review]

**Comment by alfredh** [comment]
There are 9 successful builds, and 6 failing. Here is one failing:


```
2023-06-12T12:01:58.6269477Z ##[section]Starting: *** Test Modified Ports
2023-06-12T12:01:58.6373745Z ==============================================================================
2023-06-12T12:01:58.6373995Z Task         : PowerShell
2023-06-12T12:01:58.6374076Z Description  : Run a PowerShell script on Linux, macOS, or Windows
2023-06-12T12:01:58.6374198Z Version      : 2.220.0
2023-06-12T12:01:58.6374282Z Author       : Microsoft Corporation
2023-06-12T12:01:58.6374349Z Help         : https://docs.microsoft.com/azure/devops/pipelines/tasks/utility/powershell
2023-06-12T12:01:58.6374469Z ==============================================================================
2023-06-12T12:01:59.4172490Z Generating script.
2023-06-12T12:01:59.4213211Z Formatted command: . 'C:\a\1\s\scripts\azure-pipelines\test-modified-ports.ps1' -Triplet x64-windows-static -BuildReason PullRequest -BinarySourceStub "x-azblob,https://vcpkgbinarycacheeastasia.blob.core.windows.net/cache,sp=racwl&st=2023-06-05T18:10:04Z&se=2023-11-01T00:00:00Z&spr=https&sv=2022-11-02&sr=c&sig=q%2FkRvlVVT8%2BPHxOlJaaKHMYPRwnNWP28J%2BUD%2FED7Lic%3D" -WorkingRoot D:\ -ArtifactStagingDirectory C:\a\1\a
2023-06-12T12:01:59.4519533Z ========================== Starting Command Output ===========================
2023-06-12T12:01:59.4708308Z ##[command]"C:\Program Files\PowerShell\7\pwsh.exe" -NoLogo -NoProfile -NonInteractive -ExecutionPolicy Unrestricted -Command ". 'C:\a\_temp\4d567087-c474-483e-8a44-2166408a1830.ps1'"
2023-06-12T12:01:59.8548830Z Build reason was Pull Request, using binary caching in read write mode, skipping failures.
2023-06-12T12:01:59.8944836Z 
2023-06-12T12:01:59.8953777Z     Directory: C:\a\1\s
2023-06-12T12:01:59.8957048Z 
2023-06-12T12:01:59.8963599Z [32;1mMode                 LastWriteTime         Length Name[0m
2023-06-12T12:01:59.8969743Z [32;1m----                 -------------         ------ ----[0m
2023-06-12T12:01:59.9076761Z d----           6/12/2023 12:01 PM                [44;1mempty[0m
2023-06-12T12:02:00.0416945Z Clearing contents of D:\buildtrees
2023-06-12T12:02:00.0423586Z Clearing contents of D:\installed
2023-06-12T12:02:00.0430077Z Clearing contents of D:\packages
2023-06-12T12:10:30.2383245Z vcpkg has crashed. Please create an issue at *** containing a brief summary of what you were trying to do and the following information.
2023-06-12T12:10:30.2414575Z 
2023-06-12T12:10:30.2451287Z Version=2023-06-08-19a3cb921a585fce63c0651f56a991a5a08bdc60
2023-06-12T12:10:30.2461553Z EXCEPTION=bad allocation
2023-06-12T12:10:30.2461899Z CMD=
2023-06-12T12:10:30.2462075Z C:\a\1\s\vcpkg.exe|
2023-06-12T12:10:30.2462392Z x-ci-clean|
2023-06-12T12:10:30.2462612Z --x-buildtrees-root=D:\buildtrees|
2023-06-12T12:10:30.2462875Z --x-install-root=D:\installed|
2023-06-12T12:10:30.2463085Z --x-packages-root=D:\packages|
2023-06-12T12:10:30.2463343Z --overlay-ports=scripts/test_ports|
2023-06-12T12:10:31.0081335Z ##[error][31;1mvcpkg clean failed[0m
[31;1mAt C:\a\1\s\scripts\azure-pipelines\test-modified-ports.ps1:134 char:5[0m
[31;1m+     throw "vcpkg clean failed"[0m
[31;1m+     ~~~~~~~~~~~~~~~~~~~~~~~~~~[0m
[31;1m+ CategoryInfo          : OperationStopped: (vcpkg clean failed:String) [], RuntimeException[0m
[31;1m+ FullyQualifiedErrorId : vcpkg clean failed[0m
2023-06-12T12:10:31.0147558Z ##[error]PowerShell exited with code '1'.
2023-06-12T12:10:31.0899681Z ##[section]Finishing: *** Test Modified Ports
```

**Comment by JonLiu1993** [comment]
@alfredh, I tested usage again but also failed:
````
1> [CMake] -- Configuring done (14.4s)
1> [CMake] CMake Error at F:/Feature-test/libre/vcpkg/installed/x64-windows/share/libre/libre-config.cmake:21 (set_target_properties):
1> [CMake]   The link interface of target "libre::libre" contains:
1> [CMake] 
1> [CMake]     libre::re_shared
1> [CMake] 
1> [CMake]   but the target was not found.  Possible reasons include:
1> [CMake] 
1> [CMake]     * There is a typo in the target name.
1> [CMake]     * A find_package call is missing for an IMPORTED target.
1> [CMake]     * An ALIAS target is missing.
````

**Comment by alfredh** [comment]
hi, which platform are you building for?

Please note that for Windows we only support .lib files and no .dll files:


```
if(VCPKG_TARGET_IS_WINDOWS)
    vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
endif()
```


Do you have the testprogram and command, so I can test it locally ?

**Comment by JonLiu1993** [comment]
> hi, which platform are you building for?
> 
> Please note that for Windows we only support .lib files and no .dll files:
> 
> ```
> if(VCPKG_TARGET_IS_WINDOWS)
>     vcpkg_check_linkage(ONLY_STATIC_LIBRARY)
> endif()
> ```
> 
> Do you have the testprogram and command, so I can test it locally ?


The `libre:x64-windows` I tested, the simple CMakeLists file tested is the same as last time:
https://github.com/microsoft/vcpkg/pull/31735#issuecomment-1580389138

**Comment by alfredh** [comment]
Hi, I have tested on arm64-osx and it appears to be working:




```
Alfreds-MBP-M1:tmp alfredh$ cmake -DCMAKE_TOOLCHAIN_FILE=/Users/alfredh/tmp/vcpkg/scripts/buildsystems/vcpkg.cmake .
-- The C compiler identification is AppleClang 14.0.0.14000029
-- The CXX compiler identification is AppleClang 14.0.0.14000029
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenSSL: /Users/alfredh/tmp/vcpkg/installed/arm64-osx/debug/lib/libcrypto.a (found version "3.1.1")  
-- Found ZLIB: /Users/alfredh/tmp/vcpkg/installed/arm64-osx/lib/libz.a (found version "1.2.13") 
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/alfredh/tmp


Alfreds-MBP-M1:tmp alfredh$ make
[ 50%] Building CXX object CMakeFiles/main.dir/usageTest.cpp.o
[100%] Linking CXX executable main
[100%] Built target main


Alfreds-MBP-M1:tmp alfredh$ ./main 
version 3.2.0
```




CMake file:
```
cmake_minimum_required(VERSION 3.16)
project(min-vcpkg-tiff-test)


find_package(libre CONFIG REQUIRED)

add_executable(main usageTest.cpp)

target_link_libraries(main PRIVATE libre::re)
```

usageTest.cpp:
```c
#include <re.h>


int main()
{
	libre_init();

	re_printf("version %s\n", sys_libre_version_get());
}
```

**Comment by JonLiu1993** [comment]
@alfredh, Could you please test it on regular windows platform? Such as `libre:X64-windows`? Because most users use the windows platform.

File: ports/libre/wip.patch [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
++include("${CMAKE_CURRENT_LIST_DIR}/libre-targets.cmake")
++
++# convenience target libre::libre for uniform usage
++if(NOT TARGET libre::libre)
++    if(TARGET libre::re_shared AND (BUILD_SHARED_LIBS OR NOT TARGET libre::re))
++        add_library(libre::libre INTERFACE IMPORTED)
++        set_target_properties(libre::libre PROPERTIES INTERFACE_LINK_LIBRARIES libre::re_shared)
++    elseif(TARGET libre::re AND (NOT BUILD_SHARED_LIBS OR NOT TARGET libre::re_shared))
++        add_library(libre::libre INTERFACE IMPORTED)
++        set_target_properties(libre::libre PROPERTIES INTERFACE_LINK_LIBRARIES libre::re_shared)
```
Comment (dg0yt) [inline ports/libre/wip.patch:138]: The usage error comes from this line which should use libre::re instead of libre::re_shared.

**Comment by alfredh** [comment]
thanks Kai, I updated the patch and pushed the changes.

I also tested the usage locally with triplet `x64-linux`:


```
alfredh@debian:~/tmp$ cmake -DCMAKE_TOOLCHAIN_FILE=/home/alfredh/tmp/vcpkg/scripts/buildsystems/vcpkg.cmake .
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE  
-- Found OpenSSL: /home/alfredh/tmp/vcpkg/installed/x64-linux/debug/lib/libcrypto.a (found version "3.1.0")  
-- Found ZLIB: /home/alfredh/tmp/vcpkg/installed/x64-linux/lib/libz.a (found version "1.2.13") 
-- Configuring done
-- Generating done
-- Build files have been written to: /home/alfredh/tmp
alfredh@debian:~/tmp$ make
[ 50%] Building CXX object CMakeFiles/main.dir/usageTest.cpp.o
[100%] Linking CXX executable main
[100%] Built target main
alfredh@debian:~/tmp$ ./main 
version 3.2.0
```

**Comment by JonLiu1993** [comment]
Tested usage successfully by `libre:x64-windows`:
````
libre provides CMake targets:

    find_package(libre CONFIG REQUIRED)
    target_link_libraries(main PRIVATE libre::libre)
````

**Review by JonLiu1993 — DISMISSED** [review]

File: ports/libre/vcpkg.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+  "name": "libre",
```
Comment (vicroms) [inline ports/libre/vcpkg.json:2]: At least to me, `libre` has a strong association with the LibreOffice software suite. For this reason, I suggest using `baresip-libre` as the port name.

**Review by vicroms — CHANGES_REQUESTED** [review]

File: ports/libre/vcpkg.json (hunk 4 above)
Comment (alfredh) [reply ports/libre/vcpkg.json:2]: There are also other open source projects like `libressl`.

Do you want me to rename the project and the folder and push it ?

**Comment by alfredh** [comment]
Hi,

I have renamed the project as suggested.

Is there any chance to merge this PR soon ?


Br,
Alfred

**Review by vicroms — APPROVED** [review]

**Comment by alfredh** [comment]
Great, thank you all for help and review!

**Pull request merged**